spec.html

```diff
@@ -24623,10 +25134,103 @@ <h1>Static Semantics: ContainsArguments</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-class-definitions-static-semantics-methoddecorators">
+      <h1>Static Semantics: Method Decorators</h1>
+      <emu-grammar>
+        ClassElement : DecoratorList? MethodDefinition
+      </emu-grammar>
+      <ul>
+        <li>
+          It is a Syntax Error if |DecoratorList| is present and PropName of |MethodDefinition| is `"constructor"`.
+        </li>
+      </ul>
+    </emu-clause>
+
+    <emu-clause id="sec-createfieldinitializerfunction" type="abstract operation">
+      <h1>
+        CreateFieldInitializerFunction (
+          _homeObject_: an Object,
+          _propName_: a property key or Private Name,
+          _Initializer_: a Function,
+        )
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Let _formalParameterList_ be an instance of the production <emu-grammar>FormalParameters : [empty]</emu-grammar>.
+        1. Let _scope_ be the LexicalEnvironment of the running execution context.
+        1. Let _privateScope_ be the running execution context's PrivateEnvironment.
+        1. Let _sourceText_ be the empty sequence of Unicode code points.
+        1. Let _initializer_ be ! OrdinaryFunctionCreate(%Function.prototype%, _sourceText_, _formalParameterList_, _Initializer_, ~non-lexical-this~, _scope_, _privateScope_).
+        1. Perform MakeMethod(_initializer_, _homeObject_).
+        1. Set _initializer_.[[ClassFieldInitializerName]] to _propName_.
+        1. Return _initializer_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-makeautoaccessorgetter" type="abstract operation">
+      <h1>
+        MakeAutoAccessorGetter (
+          _homeObject_: an Object,
+          _name_: a property key or Private Name,
+          _privateStateName_: a Private Name,
+        )
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Let _steps_ be the steps of an AutoAccessorGetter function as specified below.
+        1. Let _length_ be the number of non-optional parameters of an AutoAccessorGetter function as specified below.
+        1. Let _getter_ be ! CreateBuiltinFunction(_steps_, _length_, *""*, &laquo; [[PrivateStateKey]] &raquo;).
+        1. Set _getter_.[[PrivateStateKey]] to _privateStateName_.
+        1. Perform MakeMethod(_getter_, _homeObject_).
+        1. Perform SetFunctionName(_getter_, _name_, *"get"*).
+        1. Return _getter_.
+      </emu-alg>
+      <p>An AutoAccessorGetter function is an anonymous built-in function with a [[PrivateStateKey]] internal slot. When an AutoAccessorGetter function that expects no arguments is called it performs the following steps:</p>
```

Comment:
What happens when an AutoAccessorGetter _does_ expect arguments.

Reply:
i think this will get cleaned up when it's converted to use abstract closures.